distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso

```diff
@@ -201,6 +210,30 @@ type JS_Object
     to_default_visualization_data self =
         render self
 
+    ## Creates an Enso object from the JS_Object.
+    into : Any -> Any
+    into self target_type = case target_type of
+        JS_Object -> self
+        Vector -> self.to_vector
+        Map -> Map.from_pairs self.to_vector
+        _ ->
+            ## First try a conversion
+            Panic.catch No_Such_Conversion (self.to target_type) _->
+                ## If that fails, try to construct the type
+                meta = Meta.meta target_type
+                meta_type = if meta.is_a Meta.Type then meta else Meta.meta (Meta.type_of target_type)
+                constructors = meta_type.constructors
+                filtered = if self.contains_key "constructor" then constructors.filter c-> c.name == self.get "constructor" else constructors
+                case filtered.length of
+                    0 -> Error.throw (Illegal_Argument.Error "Unable to build a "+target_type.to_text+" cannot find the constructor.")
+                    1 ->
+                        constructor = filtered.at 0
+                        field_names = constructor.fields
+                        if field_names.all self.contains_key then constructor.new (field_names.map self.at) else
+                            Error.throw (Illegal_Argument.Error "Missing fields for constructor.")
+                    _ -> Error.throw (Illegal_Argument.Error "Unable to build a "+target_type.to_text+" cannot find unique constructor.")
```

Comment:
Is this case ever possible?

Reply:
Yes - something like `Filter_Condition` has lots of constructors and if the JSON doesn't have a specific one it can't work out what to build.

Reply:
Ah I forgot that `constructor` argument may be optional.